tests/Aql/AqlBuildSupervisedAqlValueTest.cpp

```diff
@@ -0,0 +1,196 @@
+#include "gtest/gtest.h"
+#include "Basics/ResourceUsage.h"
+#include "Basics/SupervisedBuffer.h"
+#include "Aql/AqlValue.h"
+#include <velocypack/Builder.h>
+#include <string>
+
+using arangodb::GlobalResourceMonitor;
+using arangodb::ResourceMonitor;
+using arangodb::ResourceUsageScope;
+using arangodb::aql::AqlValue;
+using arangodb::aql::buildSupervisedAqlValue;
+using arangodb::basics::SupervisedBuffer;
+using arangodb::velocypack::Builder;
+using arangodb::velocypack::Value;
+
+TEST(BuildSupervisedAqlValueTest,
+     AccountsMemoryLargeAndSmallValuesNormalBuffer) {
+  ResourceMonitor monitor(GlobalResourceMonitor::instance());
+
+  {
+    ResourceUsageScope usageScope(monitor);
+    AqlValue largeValue;
+    {
+      Builder builder;
+      builder.openArray();
+      builder.add(Value(std::string(1024, 'a')));
+      builder.close();
+
+      ASSERT_EQ(monitor.current(), 0);
+      largeValue = buildSupervisedAqlValue(builder, usageScope);
+
+      ASSERT_EQ(monitor.current(), largeValue.memoryUsage());
+    }
+    ASSERT_EQ(monitor.current(), largeValue.memoryUsage());
+    largeValue.destroy();
+    ASSERT_EQ(monitor.current(), 0);
+  }
+
+  {
+    ResourceUsageScope usageScope(monitor);
+    AqlValue smallValue;
+    {
+      Builder builder;
+      builder.openArray();
+      builder.add(Value(1));
+      builder.close();
+      ASSERT_EQ(monitor.current(), 0);
+      smallValue = buildSupervisedAqlValue(builder, usageScope);
+      ASSERT_EQ(monitor.current(), smallValue.memoryUsage());
+    }
+    ASSERT_EQ(monitor.current(), 0);
+    smallValue.destroy();
+    ASSERT_EQ(monitor.current(), 0);
+  }
+}
+
+TEST(BuildSupervisedAqlValueTest,
+     AccountsMemoryLargeAndSmallValuesSupervisedBuffer) {
+  ResourceMonitor monitor(GlobalResourceMonitor::instance());
+
+  {
+    ResourceUsageScope usageScope(monitor);
+    AqlValue largeValue;
+    {
+      SupervisedBuffer supervisedBuffer(monitor);
+      Builder builder(supervisedBuffer);
+      builder.openArray();
+      builder.add(Value(std::string(1024, 'a')));
+      builder.close();
+
+      ASSERT_GT(monitor.current(), 0);
+      largeValue = buildSupervisedAqlValue(builder, usageScope);
+      // While builder exists, monitor >= aql value usage (buffer + aql value)
+      ASSERT_GE(monitor.current(), largeValue.memoryUsage());
+    }
+    // now monitor is only the aql value usage
+    ASSERT_EQ(monitor.current(), largeValue.memoryUsage());
+    largeValue.destroy();
+    ASSERT_EQ(monitor.current(), 0);
+  }
+
+  {
+    ResourceUsageScope usageScope(monitor);
+    AqlValue smallValue;
+    {
+      SupervisedBuffer supervisedBuffer(monitor);
+      Builder builder(supervisedBuffer);
+      builder.openArray();
+      builder.add(Value(1));
+      builder.close();
+      ASSERT_GT(monitor.current(), 0);
```

Comment:
I do not understand this.
Why is it here GT 0
But in above comment it is equal?

This one is what I expect!